src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_golemagg.cpp

```diff
@@ -58,13 +60,27 @@ class boss_golemagg : public CreatureScript
             earthquakeTimer = 0;
             pyroblastTimer = urand(3000, 7000);
             enraged = false;
+            DoCastSelf(SPELL_MAGMASPLASH);
+            DoCastSelf(SPELL_GOLEMAGG_TRUST_AURA);
+            DoCastSelf(SPELL_DOUBLE_ATTACK);
+
+            instance->SetBossState(DATA_GOLEMAGG, FAIL);
```

Comment:
keep in mind Reset() is also called when the mob has just spawned, so this will default his boss state to FAIL

Reply:
Good to know!